File: 8kyu/alternating-case.rb
# https://www.codewars.com/kata/alternating-case-%3C-equals-%3E-alternating-case/train/ruby

# Didn't know about String#swapcase before this challenge
class String
  def to_alternating_case
    self.chars.map { |c| c =~ /[[:upper:]]/ ? c.downcase : c.upcase }.join
  end
end

File: 8kyu/alternating_case.rs
// https://www.codewars.com/kata/alternating-case-%3C-equals-%3E-alternating-case/train/rust

fn to_alternating_case(s: &str) -> String {
   let mut out = String::new();

   for c in s.chars() {
       if c == c.to_uppercase().next().unwrap() {
         out.push(c.to_lowercase().next().unwrap());
       } else {
         out.push(c.to_uppercase().next().unwrap());
       }
   }

   out
}
